### PR TITLE
Add Rust SDK to API Docs

### PR DESCRIPTION
**Pull request by agunde406**

### CODE REVIEW

File: sdk/rust/src/signing/mod.rs [hunk 1]
```diff
@@ -65,23 +69,72 @@ impl std::fmt::Display for Error {
...
+    ///
+    /// * `message`- the message bytes
+    /// * `private_key` the private key
+    ///
+    /// # Returns
+    ///
+    /// * `signature` - The signature in a hex-encoded string
     fn sign(&self, message: &[u8], key: &PrivateKey) -> Result<String, Error>;
+
+    /// Verifies that a signature of a message was produced with the
```
Comment (nick-drozd) [inline sdk/rust/src/signing/mod.rs]: a signature -> the signature

File: sdk/rust/src/signing/mod.rs [hunk 2]
```diff
@@ -65,23 +69,72 @@ impl std::fmt::Display for Error {
...
+    /// * `message` - the message bytes
+    /// * `public_key` - the public key to use for verification
+    ///
+    /// # Returns
+    ///
+    /// * `boolean` - True if the public key is associated with the signature for that method,
+    ///            False otherwise
     fn verify(&self, signature: &str, message: &[u8], key: &PublicKey) -> Result<bool, Error>;
+
+    /// Produce a public key for the given private key.
```
Comment (nick-drozd) [inline sdk/rust/src/signing/mod.rs]: a public key -> the public key

(There can be only one, right?)

File: ci/sawtooth-build-docs [hunk 3]
```diff
@@ -36,8 +36,10 @@ RUN echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sou
     git \
     libffi-dev \
     libssl-dev \
+    libzmq3-dev \
```
Comment (nick-drozd) [inline ci/sawtooth-build-docs]: Is this related?

**Comment by nick-drozd** [comment]
Looks like you have some rebase weirdness with one of Peter's commits.

File: ci/sawtooth-build-docs (hunk 3 above)
Comment (agunde406) [reply ci/sawtooth-build-docs]: Yes, It is needed for the rust to build correctly

File: sdk/rust/src/signing/mod.rs [hunk 4]
```diff
@@ -24,8 +24,12 @@ use std::borrow::Borrow;
...
 #[derive(Debug)]
 pub enum Error {
+    /// Returned when trying to create an algorithm which does not exist.
     NoSuchAlgorithm(String),
+    /// Returned when an error occurs during deserialization of a Private or
+    /// Public key from various formats.
     ParseError(String),
+    /// Returned when an error occurs during the signing process.
     SigningError(Box<StdError>),
     KeyGenError(String),
```
Comment (peterschwarz) [inline sdk/rust/src/signing/mod.rs]: Missed a doc here.

File: sdk/rust/src/signing/mod.rs [hunk 5]
```diff
@@ -65,23 +69,72 @@ impl std::fmt::Display for Error {
...
+    /// `private_key` - a private key
+    ///
+    /// # Returns
+    /// * `public_key` - the public key for the given private key
     fn get_public_key(&self, private_key: &PrivateKey) -> Result<Box<PublicKey>, Error>;
+
+    ///Generates a new random PrivateKey using this context.
+    /// # Returns
+    ///
+    /// *`private_key` - a random private key
```
Comment (peterschwarz) [inline sdk/rust/src/signing/mod.rs]: add space after `*`
